exp/api/v1beta1/azuremachinepool_webhook.go

```diff
@@ -203,6 +199,22 @@ func (amp *AzureMachinePool) ValidateSystemAssignedIdentity(old runtime.Object)
 	}
 }
 
+// ValidateSystemAssignedIdentityRole validates the scope and roleDefinitionID for the system-assigned identity.
+func (amp *AzureMachinePool) ValidateSystemAssignedIdentityRole() error {
```

Comment:
same comments about validating name isn't set in both places here - consider how we might refactor this to share validation code with AzureMachine webhook